bundles/python/python_lexer.moon

```diff
@@ -100,4 +140,14 @@ howl.aux.lpeg_lexer ->
       identifier,
       decorator
     }
+
+    f_string: any f_strings
+
+    string: any {
+      f_prefix * V'f_string'
+      string
+    }
   }
+
+  rules[k] = v for k, v in pairs interpolations
```

Comment:
One way around having to explode all of these as separate rules is to capture the actual opening quote in a named group capture, and then match the correct ending quote using `match_back`. This used for instance in the Ruby lexer, where you can specify your own quote character e.g. for a double quoted string, and in the Lua lexer for where block quotes can have an arbitrary number of '='s. This would allow for a single body of rules for all the fstring cases, which would probably be easier to follow than having to expand the variants in one's head. I'll leave it up to you whether to pursue this at this time or not, but you might be pleasantly surprised (would take of the ordering issue at the same time as the opening quote list would be ordered).